upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -204,3 +207,30 @@ func (tf *TemplateFunctions) ExternalDnsArgv() ([]string, error) {
 
 	return argv, nil
 }
+
+func (tf *TemplateFunctions) ProxyEnv() (map[string]string, error) {
+	envs := map[string]string{}
+	proxies := tf.cluster.Spec.EgressProxy
+	if proxies == nil {
+		return envs, nil
+	}
+	httpProxy := proxies.HTTPProxy
+	if httpProxy.Host != "" {
+		url := "http://"
+		if httpProxy.User != "" {
+			url += httpProxy.User
+			if httpProxy.Password != "" {
+				url += ":" + httpProxy.Password
+			}
+			url += "@"
+		}
+		url += httpProxy.Host + ":" + strconv.Itoa(httpProxy.Port)
```

Comment:
In practice, it rarely works this way, and ports are nearly always specified in proxy configurations.  Both http_proxy and https_proxy work over scheme http to connect to the proxy (typically, some digging tells me https for client-to-proxy is possible but I am not sure how common it is in practice)  For example, curl will, if the port is ommited, assume port 1080.  wget is undocumented (seems it might use the default for the url scheme in the proxy), and likewise with golang net/http. I see two possible approaches: We could check check for nil, making the port optional in the configuration and here leaving colon and port off if it was omitted in the cluster spec, which would leave behavior up to the specific client and fate.  The other option is to require an explicit port for a valid configuration.